src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -436,9 +436,19 @@ private String getPrincipalName(String username, String domainName) {
                 switch (groupLookupStrategy) {
                 case AUTO:
                     // try the accurate one first, and if it's too slow fall back to recursive in the hope that it's faster
-                    long start = System.currentTimeMillis();
-                    boolean found = chainGroupLookup(domainDN, userDN, context, groups);
-                    long duration = (System.currentTimeMillis() - start) / TimeUnit2.SECONDS.toMillis(1);
+                    long start = System.nanoTime();
+                    boolean found = false;
+                    long duration = 0;
+                    try {
+                        found = chainGroupLookup(domainDN, userDN, context, groups);
+                        duration = TimeUnit2.NANOSECONDS.toSeconds(System.nanoTime() - start);
+                    } catch (NamingException e) {
+                        if (e.getMessage().contains("LDAP response read timed out")) {
+                            LOGGER.log(Level.WARNING, "LDAP response read time out. AD will fall back to recursive lookup", e);
+                        } else {
+                            throw e;
+                        }
+                    }
                     if (!found || duration >= 10) {
```

Comment:
Not this PR - but this logic smells wrong to me.  we could have taken longer than 10s to get all the results - but that may not be due to a read timeout.